src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_pv_power_formula.py

```diff
@@ -58,18 +63,79 @@ def generate(  # noqa: DOC502
             # frequency as the other streams.  So we subscribe with a non-existing
             # component id, just to get a `None` message at the resampling interval.
             builder.push_component_metric(
-                NON_EXISTING_COMPONENT_ID, nones_are_zeros=True
+                NON_EXISTING_COMPONENT_ID,
+                nones_are_zeros=True,
             )
             return builder.build()
 
-        for idx, component in enumerate(pv_components):
-            if idx > 0:
-                builder.push_oper("+")
+        if self._config.allow_fallback:
+            fallbacks = self._get_fallback_formulas(pv_components)
 
-            # should only be the case if the component is not a meter
-            builder.push_component_metric(
-                component.component_id,
-                nones_are_zeros=component.category != ComponentCategory.METER,
-            )
+            for idx, (primary_component, fallback_formula) in enumerate(
+                fallbacks.items()
+            ):
+                if idx > 0:
+                    builder.push_oper("+")
+
+                # should only be the case if the component is not a meter
+                builder.push_component_metric(
+                    primary_component.component_id,
+                    nones_are_zeros=False,
+                    fallback=fallback_formula,
+                )
```

Comment:
No reason to think the primary component would always be a meter.

```suggestion
                builder.push_component_metric(
                    primary_component.component_id,
                    nones_are_zeros=True,
                    fallback=fallback_formula,
                )

Reply:
Done